Sources/Sentry/SentrySystemEventsBreadcrumbs.m

```diff
@@ -0,0 +1,117 @@
+    // https://developer.apple.com/documentation/uikit/uidevicebatterystatedidchangenotification
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(batteryStateChanged:) name:UIDeviceBatteryStateDidChangeNotification object:currentDevice];
+    
+    // for testing only
+//    [[NSNotificationCenter defaultCenter] postNotificationName:@"UIDeviceBatteryStateDidChangeNotification" object:currentDevice];
+//    [[NSNotificationCenter defaultCenter] postNotificationName:@"UIDeviceOrientationDidChangeNotification" object:currentDevice];
+}
+
+- (void)batteryStateChanged:(NSNotification*)notification
+{
+    UIDevice *currentDevice = notification.object;
+    
+    // Notifications for battery level change are sent no more frequently than once per minute
+    NSDictionary* batteryData = [self getBatteryStatus:currentDevice];
+    [batteryData setValue:@"BATTERY_STATE_CHANGE" forKey:@"action"];
+    
+    SentryBreadcrumb *crumb =
+        [[SentryBreadcrumb alloc] initWithLevel:kSentryLevelInfo
+                                       category:@"device.event"];
+    crumb.type = @"system";
+    crumb.data = batteryData;
+    [SentrySDK addBreadcrumb:crumb];
+}
+
+- (NSDictionary*)getBatteryStatus:(UIDevice*)currentDevice
+{
+    // borrowed and adapted from https://github.com/apache/cordova-plugin-battery-status/blob/master/src/ios/CDVBattery.m
+    UIDeviceBatteryState currentState = [currentDevice batteryState];
+
+    BOOL isPlugged = NO; // UIDeviceBatteryStateUnknown or UIDeviceBatteryStateUnplugged
+    if ((currentState == UIDeviceBatteryStateCharging) || (currentState == UIDeviceBatteryStateFull)) {
+        isPlugged = YES;
+    }
+    float currentLevel = [currentDevice batteryLevel];
+    
+    NSMutableDictionary *batteryData = [NSMutableDictionary new];
+
+    // W3C spec says level must be null if it is unknown
+    if ((currentState != UIDeviceBatteryStateUnknown) || (currentLevel != -1.0)) {
+        float w3cLevel = (currentLevel * 100);
+        [batteryData setValue:[NSNumber numberWithFloat:w3cLevel] forKey:@"level"];
+    } else {
+        [SentryLog logWithMessage:@"batteryLevel is unknown."
+                         andLevel:kSentryLogLevelDebug];
+    }
+    [batteryData setValue:[NSNumber numberWithBool:isPlugged] forKey:@"plugged"];
+
+    return batteryData;
+}
+
+- (void)initOrientationObserver:(UIDevice*)currentDevice
+{
+    if (currentDevice.isGeneratingDeviceOrientationNotifications == NO) {
+        [currentDevice beginGeneratingDeviceOrientationNotifications];
+    }
+    // https://developer.apple.com/documentation/uikit/uideviceorientationdidchangenotification
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(batteryStateChanged:) name:UIDeviceOrientationDidChangeNotification object:currentDevice];
```

Comment:
@HazAT ideas why this is never called? does it require something special lifecycle to be called?